Whiskey/Tasks/Invoke-WhiskeyExec.ps1

```diff
@@ -120,19 +116,87 @@ function Invoke-WhiskeyExec
     }
 
 
-    $successExitCode = 0
+    [int[]]$successExitCode = $null
+    $successExitRangeGreaterThan = $null
+    $successExitRangeLessThan = $null
+
     if ( $TaskParameter['SuccessExitCode'] )
     {
-        $successExitCode = $TaskParameter['SuccessExitCode']
+        $TaskParameter['SuccessExitCode'] | ForEach-Object {

```

Comment:
Move this whole block to after the process runs. Evaluate the exit code after the process exits. If you move it there, the logic for evaluating can be simpler. As soon as you find a success exit code matches the exit code, you can return from the function. 

Here's the code I came up with:

```
$exitCode = $process.ExitCode

$successExitCodes = $TaskParameter['SuccessExitCode']
if( -not $successExitCodes )
{
    $successExitCodes = '0'
}

foreach( $successExitCode in $successExitCodes )
{
    if( $successExitCode -match '^(\d)+$' )
    {
        if( $exitCode -eq [int]$Matches[1] )
        {
            Write-Verbose -Message ('  Exit code {0} = {1}' -f $exitCode,$Matches[1])
            return
        }
    }

    if( $successExitCode -match '^(<|<=|>=|>)\s*(\d)+$' )
    {
        $operator = $Matches[1]
        $successExitCode = [int]$Matches[2]
        switch( $operator )
        {
            '<'
            {
                if( $exitCode -lt $successExitCode )
                {
                    Write-Verbose -Message ('  Exit code {0} < {1}' -f $exitCode,$successExitCode)
                    return
                }
            }
            '<=' 
            {
                if( $exitCode -le $successExitCode )
                {
                    Write-Verbose -Message ('  Exit code {0} <= {1}' -f $exitCode,$successExitCode)
                    return
                }
            }
            '>='
            {
                if( $exitCode -ge $successExitCode )
                {
                    Write-Verbose -Message ('  Exit code {0} >= {1}' -f $exitCode,$successExitCode)
                    return
                }
            }
            '>' 
            {
                if( $exitCode -gt $successExitCode )
                {
                    Write-Verbose -Message ('  Exit code {0} > {1}' -f $exitCode,$successExitCode)
                    return
                }
            }
        }
    }

    if( $successExitCode -match '^(\d)+\.\.\(d+)$' )
    {
        if( [int]$Matches[1] -le $exitCode -and $exitCode -le [int]$Matches[2] )
        {
            Write-Verbose -Message ('  Exit code {0} <= {1} <= {2}' -f $Matches[1],$exitCode,$Matches[2])
            return
        }
    }
}
```